### ISSUE TITLE
Unhandled Exception

### ISSUE THREAD

**Issue by gitagogaming** [terrymacdonald/DisplayMagician#24]
Received this error when switching my 'profiles'.. all appears ok after it completed but i figured this may be something you'd want to know about.. 

See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

![image](https://user-images.githubusercontent.com/76603653/124512425-5678d080-dda6-11eb-946c-48ae115bf051.png)

```
************** Exception Text **************
WindowsDisplayAPI.Exceptions.TargetNotAvailableException: Extra information about the target is not available.
   at WindowsDisplayAPI.DisplayConfig.PathDisplayTarget.get_DevicePath()
   at DisplayMagicianShared.Topology.PathTarget..ctor(PathTargetInfo targetInfo, SurroundTopology surround) in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\Topology\PathTarget.cs:line 13
   at DisplayMagicianShared.Topology.Path.<>c.<.ctor>b__0_0(PathTargetInfo targetDisplay) in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\Topology\Path.cs:line 20
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at DisplayMagicianShared.Topology.Path..ctor(PathInfo pathInfo) in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\Topology\Path.cs:line 20
   at DisplayMagicianShared.ProfileRepository.<>c.<UpdateActiveProfile>b__34_0(PathInfo info) in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:line 544
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at DisplayMagicianShared.ProfileRepository.UpdateActiveProfile() in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:line 541
   at DisplayMagician.UIForms.DisplayProfileForm.WndProc(Message& m) in H:\vscode-projects\DisplayMagician\DisplayMagician\UIForms\DisplayProfileForm.cs:line 455
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4360.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
DisplayMagician
    Assembly Version: 1.1.4.0
    Win32 Version: 1.1.4.0
    CodeBase: file:///C:/Program%20Files/DisplayMagician/DisplayMagician.exe
----------------------------------------
NLog
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.10.13013
    CodeBase: file:///C:/Program%20Files/DisplayMagician/NLog.DLL
----------------------------------------
DisplayMagician.Shared
    Assembly Version: 1.1.4.0
    Win32 Version: 1.1.4.0
    CodeBase: file:///C:/Program%20Files/DisplayMagician/DisplayMagician.Shared.DLL
----------------------------------------
McMaster.Extensions.CommandLineUtils
    Assembly Version: 3.1.0.0
    Win32 Version: 3.1.0.0
    CodeBase: file:///C:/Program%20Files/DisplayMagician/McMaster.Extensions.CommandLineUtils.DLL
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4341.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4360.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4190.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4360.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Windows.UI
    Assembly Version: 255.255.255.255
    Win32 Version: 10.0.10011.16384
    CodeBase: file:///C:/WINDOWS/system32/WinMetadata/Windows.UI.winmd
----------------------------------------
System.Runtime
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Runtime.dll
----------------------------------------
Newtonsoft.Json
    Assembly Version: 13.0.0.0
    Win32 Version: 13.0.1.25517
    CodeBase: file:///C:/Program%20Files/DisplayMagician/Newtonsoft.Json.DLL
----------------------------------------
System.Numerics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------
System.Runtime.Serialization
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4250.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
System.ServiceModel
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4250.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.ServiceModel/v4.0_4.0.0.0__b77a5c561934e089/System.ServiceModel.dll
----------------------------------------
System.Runtime.InteropServices.RuntimeInformation
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.InteropServices.RuntimeInformation/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Runtime.InteropServices.RuntimeInformation.dll
----------------------------------------
System.ComponentModel.DataAnnotations
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.ComponentModel.DataAnnotations/v4.0_4.0.0.0__31bf3856ad364e35/System.ComponentModel.DataAnnotations.dll
----------------------------------------
System.ValueTuple
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.ValueTuple/v4.0_4.0.0.0__cc7b13ffcd2ddd51/System.ValueTuple.dll
----------------------------------------
System.Xml.Linq
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml.Linq/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.Linq.dll
----------------------------------------
System.Web
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4330.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.Web/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Web.dll
----------------------------------------
System.ServiceModel.Internals
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4250.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.ServiceModel.Internals/v4.0_4.0.0.0__31bf3856ad364e35/System.ServiceModel.Internals.dll
----------------------------------------
SMDiagnostics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4250.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/SMDiagnostics/v4.0_4.0.0.0__b77a5c561934e089/SMDiagnostics.dll
----------------------------------------
System.IdentityModel
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4250.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.IdentityModel/v4.0_4.0.0.0__b77a5c561934e089/System.IdentityModel.dll
----------------------------------------
ValveKeyValue
    Assembly Version: 0.3.1.152
    Win32 Version: 0.3.1.152
    CodeBase: file:///C:/Program%20Files/DisplayMagician/ValveKeyValue.DLL
----------------------------------------
netstandard
    Assembly Version: 2.0.0.0
    Win32 Version: 4.8.4084.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/netstandard/v4.0_2.0.0.0__cc7b13ffcd2ddd51/netstandard.dll
----------------------------------------
NHotkey
    Assembly Version: 2.0.0.0
    Win32 Version: 2.1.0.0
    CodeBase: file:///C:/Program%20Files/DisplayMagician/NHotkey.DLL
----------------------------------------
Microsoft.Toolkit.Uwp.Notifications
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.2.1
    CodeBase: file:///C:/Program%20Files/DisplayMagician/Microsoft.Toolkit.Uwp.Notifications.DLL
----------------------------------------
Windows.Foundation
    Assembly Version: 255.255.255.255
    Win32 Version: 10.0.10011.16384
    CodeBase: file:///C:/WINDOWS/system32/WinMetadata/Windows.Foundation.winmd
----------------------------------------
System.Runtime.InteropServices.WindowsRuntime
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.InteropServices.WindowsRuntime/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Runtime.InteropServices.WindowsRuntime.dll
----------------------------------------
NHotkey.WindowsForms
    Assembly Version: 2.0.0.0
    Win32 Version: 2.1.0.0
    CodeBase: file:///C:/Program%20Files/DisplayMagician/NHotkey.WindowsForms.DLL
----------------------------------------
Windows.Data
    Assembly Version: 255.255.255.255
    Win32 Version: 10.0.10011.16384
    CodeBase: file:///C:/WINDOWS/system32/WinMetadata/Windows.Data.winmd
----------------------------------------
IconLib
    Assembly Version: 0.73.0.0
    Win32 Version: 0.73.0.0
    CodeBase: file:///C:/Program%20Files/DisplayMagician/IconLib.DLL
----------------------------------------
IconExtractor
    Assembly Version: 1.0.3.0
    Win32 Version: 1.0.3.0
    CodeBase: file:///C:/Program%20Files/DisplayMagician/IconExtractor.DLL
----------------------------------------
MintPlayer.IconUtils
    Assembly Version: 1.0.4.0
    Win32 Version: 1.0.4.0
    CodeBase: file:///C:/Program%20Files/DisplayMagician/MintPlayer.IconUtils.DLL
----------------------------------------
System.Drawing.Common
    Assembly Version: 4.0.0.1
    Win32 Version: 4.6.26919.02
    CodeBase: file:///C:/Program%20Files/DisplayMagician/System.Drawing.Common.DLL
----------------------------------------
WinCopies.WindowsAPICodePack.Win32Native
    Assembly Version: 2.0.0.6
    Win32 Version: 2.0.0.6
    CodeBase: file:///C:/Program%20Files/DisplayMagician/WinCopies.WindowsAPICodePack.Win32Native.DLL
----------------------------------------
NvAPIWrapper
    Assembly Version: 0.8.1.100
    Win32 Version: 0.8.1.100
    CodeBase: file:///C:/Program%20Files/DisplayMagician/NvAPIWrapper.DLL
----------------------------------------
WindowsDisplayAPI
    Assembly Version: 1.3.0.13
    Win32 Version: 1.3.0.13
    CodeBase: file:///C:/Program%20Files/DisplayMagician/WindowsDisplayAPI.DLL
----------------------------------------
AudioSwitcher.AudioApi.CoreAudio
    Assembly Version: 4.0.0.308
    Win32 Version: 4.0.0.308
    CodeBase: file:///C:/Program%20Files/DisplayMagician/AudioSwitcher.AudioApi.CoreAudio.DLL
----------------------------------------
AudioSwitcher.AudioApi
    Assembly Version: 4.0.0.308
    Win32 Version: 4.0.0.308
    CodeBase: file:///C:/Program%20Files/DisplayMagician/AudioSwitcher.AudioApi.DLL
----------------------------------------
AutoUpdater.NET
    Assembly Version: 1.6.4.0
    Win32 Version: 1.6.4.0
    CodeBase: file:///C:/Program%20Files/DisplayMagician/AutoUpdater.NET.DLL
----------------------------------------
Microsoft.CSharp
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/Microsoft.CSharp/v4.0_4.0.0.0__b03f5f7f11d50a3a/Microsoft.CSharp.dll
----------------------------------------
System.Dynamic
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Dynamic/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Dynamic.dll
----------------------------------------
Anonymously Hosted DynamicMethods Assembly
    Assembly Version: 0.0.0.0
    Win32 Version: 4.8.4360.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/mscorlib/v4.0_4.0.0.0__b77a5c561934e089/mscorlib.dll
----------------------------------------
ImageListView
    Assembly Version: 13.8.2.0
    Win32 Version: 13.8.2.0
    CodeBase: file:///C:/Program%20Files/DisplayMagician/ImageListView.DLL
----------------------------------------
EDIDParser
    Assembly Version: 1.2.0.0
    Win32 Version: 1.2.0.0
    CodeBase: file:///C:/Program%20Files/DisplayMagician/EDIDParser.DLL
----------------------------------------
Accessibility
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll
----------------------------------------
BootMeUp
    Assembly Version: 1.2.0.0
    Win32 Version: 1.2.0.0
    CodeBase: file:///C:/Program%20Files/DisplayMagician/BootMeUp.DLL
----------------------------------------
CircularProgressBar
    Assembly Version: 2.8.0.16
    Win32 Version: 2.8.0.16
    CodeBase: file:///C:/Program%20Files/DisplayMagician/CircularProgressBar.DLL
----------------------------------------
WinFormAnimation
    Assembly Version: 1.6.0.4
    Win32 Version: 1.6.0.4
    CodeBase: file:///C:/Program%20Files/DisplayMagician/WinFormAnimation.DLL
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.



```

**Comment by terrymacdonald** [comment]
Huh. This one is weird. The error is coming from the WindowsDisplayAPI which is the library that I use for changing the displays within Windows. Not sure what could be causing that, as the exception is showing that it's faulting somewhere.

Could you please generate a TRACE DisplayMagician.log file using the instructions below, and upload it here for me to review? Also please upload your DisplayProfiles_1.0.json and Shortcuts_1.0.json files too (they are in the Profiles and Shortcuts subdirectories near to where you will find the DisplayMagician.log file.

NOTE: The DisplayMagician.log file is unfortunately just the default DisplayMagician.log file. If you open it you'll see that it only has 'WARN' entries in it. I need the TRACE entries in it too.

To generate a DisplayMagician.log file with TRACE entries in it, please do the following:

- Press WindowsKey + R to bring up a Run prompt
- Click the Browse button
- Navigate to `C:\Program Files\DisplayMagician\DisplayMagician.exe`, and click the Open button
- Go to the end of the command, and add a space, and then add `--trace` on the end.
- The whole command should read `"C:\Program Files\DisplayMagician\DisplayMagician.exe" --trace`
- Press the Run button.

This will force DisplayMagician to start logging TRACE commands until DisplayMagician is closed down.

Now. perform the actions that don't work. In your case, please open up the Display Profile window, and then close it down, and then open up the Shortcut Library and then edit a Shortcut (but don't change anything). Then close down DisplayMagician.

Then straight away please zip up the DisplayMagician.log file and upload it here with the DisplayProfiles_1.0.json and Shortcuts_1.0.json files. Please make sure that it has TRACE commands in the log file, because then we know that the --trace option worked properly.

Once I have the DisplayMagician.log file with TRACE commands in it I'll be able to see if there is anything wrong.

**Comment by terrymacdonald** [comment]
@gitagogaming I've researched this as much as I can, and I've decided I won't fix this particular issue. The reason is that all the display manipulation code is changing as part of the improvements being made to the next release, DisplayMagician v2.0.0. This will record as much information as possible about a display layout, and will have completely new NVIDIA support as well. 

As all of the code will be replaced anyway, it's not worth me fixing this at this moment. I will instead spend the time developing v2.0.0 which should make this issue go away.

Thanks
Terry

**Comment by gitagogaming** [comment]
> @gitagogaming I've researched this as much as I can, and I've decided I won't fix this particular issue. The reason is that all the display manipulation code is changing as part of the improvements being made to the next release, DisplayMagician v2.0.0. This will record as much information as possible about a display layout, and will have completely new NVIDIA support as well.
> 
> As all of the code will be replaced anyway, it's not worth me fixing this at this moment. I will instead spend the time developing v2.0.0 which should make this issue go away.
> 
> Thanks
> Terry

Alright, sounds good.. i appreciate you taking the time and looking into it...  
technically it does work just fine it just throws that error every so often.. I believe this is why it takes roughly a minute for it to complete its transformation.

**Comment by terrymacdonald** [comment]
Hi @gitogaming,

I've managed to get the windows only (no NVIDIA) screen change down to 3.5
seconds, so it will be interesting to see how quick I can get the NVIDIA
change down to. I'm working on AMD next though, and then NVIDIA after that.

Please keep an eye out for the v2.0.0 update. The upgrade will be manual as
v2.0.0 will require everyone to recreate their display profiles. I don't
want to

I think I'll create a v1.1.5 just to announce that there is a V2.0.0 when
it's ready, as not everyone visits the website. But that's a later problem
😁.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Great news! DisplayMagician v2.0.1 has been released, and it should fix this very issue. Please test it out, and if you still have any issues please log a new issue. You can get DIsplayMagician v2.0.1 from here: https://github.com/terrymacdonald/DisplayMagician/releases/tag/v2.0.1

I'm closing this issue as the fix is in DisplayMagician v2.0.1, and I need to clear out the completed issues so I am able to work on the new issues that come in.

Thanks
Terry
